builder/llpcBuilderImplInOut.cpp

```diff
@@ -848,17 +890,44 @@ Instruction* BuilderImplInOut::CreateWriteBuiltInOutput(
                              &*GetInsertPoint()));
 }
 
+// =====================================================================================================================
+// Get the type of a built-in. This overrides the one in Builder to additionally recognize the internal built-ins.
+Type* BuilderImplInOut::GetBuiltInTy(
+    BuiltInKind   builtIn,            // Built-in kind
+    InOutInfo     inOutInfo)          // Extra input/output info (shader-defined array size)
+{
+    switch (static_cast<uint32_t>(builtIn))
+    {
+    case BuiltInSamplePosOffset:
+    case BuiltInInterpLinearCenter:
+        return VectorType::get(getFloatTy(), 2);
+    case BuiltInInterpPullMode:
+        return VectorType::get(getFloatTy(), 3);
+    default:
+        return Builder::GetBuiltInTy(builtIn, inOutInfo);
+    }
+}
+
 // =====================================================================================================================
 // Get name of built-in
 StringRef BuilderImplInOut::GetBuiltInName(
     BuiltInKind   builtIn)            // Built-in type, one of the BuiltIn* constants
 {
-    switch (builtIn)
+    switch (static_cast<uint32_t>(builtIn))
     {
 #define BUILTIN(name, number, out, in, type) \
     case BuiltIn ## name: return # name;
 #include "llpcBuilderBuiltIns.h"
```

Comment:
why not adding the new built-ins in llpcBuilderBuiltIns.h?

Reply:
Because they are internal, not part of the Builder interface for use by the frontend.